Mega menu: Cache Wagtail version of menu

virginiacc opened the pull request:
Adds caching for Wagtail version of mega menu so its page queries don't impact initial page load times.

## Additions

- New template fragment cache
- Caching for Wagtail version of menu

## Testing

1. Check out branch
2. Create cache table: `python cfgov/manage.py createcachetable default_fragment_cache`
3. Run the server with Django Debug Toolbar & caching enabled: `ENABLE_DEBUG_TOOLBAR=1 ENABLE_DEFAULT_FRAGMENT_CACHE=1 ./runserver.sh`
4. Turn on `WAGTAIL_MENU` flag
5. Load [http://localhost:8000/](http://localhost:8000/) and note the large number of queries in the debug toolbar
6. Reload [http://localhost:8000/](http://localhost:8000/)  and note the number of queries has decreased as the menu is loaded from cache
7. Save a change to a Menu Item.  Reload [http://localhost:8000/](http://localhost:8000/) and verify that the cache has been cleared and the updated menu item is displayed

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

chosak commented on cfgov/v1/wagtail_hooks.py:
```diff
@@ -219,3 +221,14 @@ class MegaMenuModelAdmin(ModelAdmin):
+@receiver(post_save, sender=MegaMenuItem)
+def clear_mega_menu_cache(sender, instance, **kwargs):
+    from django.core.cache import caches
+    try:
```
What's the rationale for this `try/except`? Is it to handle the case where the cache isn't defined?

To simplify, how would you feel about always defining the cache in local settings, but defaulting it to use [`DummyCache`](https://docs.djangoproject.com/en/2.0/topics/cache/#dummy-caching-for-development) unless the environment variable is set? This would simplify the application logic.

virginiacc commented:
@chosak @richaagarwal I renamed the cache to `default_fragment_cache` and removed the try/except block in https://github.com/cfpb/cfgov-refresh/pull/4402/commits/1b525a55ee9b5e73ecd4e6afdfdaa08872d50057.

richaagarwal commented:
Before this is merged/deployed, we'll want to run `python cfgov/manage.py createcachetable default_fragment_cache` on build, staging, and production.  Not sure what the state of beta is?

virginiacc commented:
Thanks, @richaagarwal & @chosak!
